src/testing/builderbot/simulator/example_builderbot_advance.argos.in

```diff
@@ -0,0 +1,151 @@
+      />
+    </lua_controller>
+
+    <lua_controller id="block">
+      <actuators>
+        <directional_leds implementation="default" />
+        <radios implementation="default"/>
+      </actuators>
+      <sensors>
+        <radios implementation="default" show_rays="true"/>
+      </sensors>
+      <params script="@CMAKE_BINARY_DIR@/testing/builderbot/example_block.lua" />
+    </lua_controller>
+  </controllers>
+
+  <!-- *********************** -->
+  <!-- * Arena configuration * -->
+  <!-- *********************** -->
+  <arena size="3, 3, 1" center="0, 0, 0">
+    <!-- add the arena walls -->
+    <box id="north" size="0.025,2.975,0.075" movable="false">
+      <body position="1.4875,0.0125,0" orientation="0,0,0" />
+    </box>
+    <box id="east" size="2.975,0.025,0.075" movable="false">
+      <body position="0.0125,-1.4875,0" orientation="0,0,0" />
+    </box>
+    <box id="south" size="0.025,2.975,0.075" movable="false">
+      <body position="-1.4875,-0.0125,0" orientation="0,0,0" />
+    </box>
+    <box id="west" size="2.975,0.025,0.075" movable="false">
+      <body position="-0.0125,1.4875,0" orientation="0,0,0" />
+    </box>
+
+    <block id="block11">
+      <body position="0.10,0,0" orientation="0,0,0"/>
+      <controller config="block"/>
+    </block>
+
+    <block id="block12">
+      <body position="0.10,0.3,0" orientation="0,0,0"/>
+      <controller config="block"/>
+    </block>
+
+    <builderbot id="builderbot1">
+      <body position="-0.25,0,0" orientation="0,0,0"/>
+      <controller config="builderbot"/>
+    </builderbot>
+
+    <builderbot id="builderbot2">
+      <body position="-0.25,0.3,0" orientation="0,0,0"/>
+      <controller config="builderbot"/>
+    </builderbot>
+
+    <block id="block1">
+      <body position="-0.255,-0.1,0" orientation="0,0,0"/>
+      <controller config="block"/>
+    </block>
+
+    <block id="block2">
+      <body position="-0.2,-0.1,0" orientation="0,0,0"/>
+      <controller config="block"/>
+    </block>
+
+    <block id="block3">
+      <body position="-0.145.,-0.1,0" orientation="0,0,0"/>
+      <controller config="block"/>
+    </block>
+  </arena>
+
+  <!-- ******************* -->
+  <!-- * Physics engines * -->
+  <!-- ******************* -->
+  <physics_engines>
+    <dynamics3d id="dyn3d" iterations="25" default_friction="1">
+      <gravity g="9.8" />
+      <floor friction="1"/>
+      <virtual_magnetism />
+    </dynamics3d>
+  </physics_engines>
+
+  <!-- ********* -->
+  <!-- * Media * -->
+  <!-- ********* -->
+  <media>
+    <directional_led id="directional_leds" index="grid" grid_size="20,20,20"/>
+    <tag id="tags" index="grid" grid_size="20,20,20" />
+    <radio id="nfc" index="grid" grid_size="20,20,20" />
+    <radio id="wifi" index="grid" grid_size="20,20,20" />
+  </media>
+
+  <!-- ****************** -->
+  <!-- * Visualization * -->
+  <!-- ****************** -->
+  <visualization>
+    <qt-opengl lua_editor="true" show_boundary="false">
```

Comment:
Disable the Lua editor

Reply:
I hesitated about this. The reason I did this is that I intended to create an example to show how to debug. Making lua editor show up and making "data" a global variable would make user to see what is happening. I can remove these if you think it is unnecessary.

Reply:
I would leave it out since the Lua editor itself is very buggy, we can add a section called debugging in the README.md where we describe these approaches.